### PR TITLE
Enable PEP service for non-registered users as well as registered users

### PR DESCRIPTION
**Pull request by pervezroy**
Enable PEP service for non-registered users as well as registered users

### CODE REVIEW

File: i18n/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/bin/
```
Comment (guusdk) [inline i18n/.gitignore:1]: This change is probably not needed?

File: xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubEngine.java [hunk 2]
```diff
@@ -48,6 +64,12 @@
 
     private static final Logger Log = LoggerFactory.getLogger(PubSubEngine.class);
 
+	public static final SystemProperty<Boolean> PUBSUB_NONREGUSER = SystemProperty.Builder.ofType(Boolean.class)
+		    .setKey("xmpp.pubsub.createnode.nonregistereduser")
```
Comment (guusdk) [inline xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubEngine.java:68]: in `openfire_i18n.properties`, add a line like this:

`system_property.xmpp.pubsub.createnode.nonregistereduser=Allows PubSub nodes to be created for JIDs that do not represent an existing user.`

File: i18n/.gitignore (hunk 1 above)
Comment (pervezroy) [reply i18n/.gitignore:1]: agree .. probably appeared automatically by the merge process

File: xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubEngine.java (hunk 2 above)
Comment (pervezroy) [reply xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubEngine.java:68]: updated the i18.properties file with description

File: i18n/.gitignore (hunk 1 above)
Comment (guusdk) [reply i18n/.gitignore:1]: It's still there though. :)

Comment (pervezroy) [reply i18n/.gitignore:1]: removed

**Comment by guusdk** [comment]
We've agreed that this change will be pushed to the 4.5 branch (to be part of release 4.5.3), but will not be brought to 4.6 or any other future releases (other than those from the 4.5 branch).

**Pull request merged**